fedimint-core/src/admin_client.rs

```diff
@@ -60,15 +73,58 @@ pub struct PeerServerParams {
     /// Name of the peer, used in TLS auth
     pub name: String,
     /// Status of the peer if known
-    pub status: Option<ServerStatus>,
+    pub status: Option<ServerStatusLegacy>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct SetLocalParamsRequest {
+    /// Name of the peer, used in TLS auth
+    pub name: String,
+    /// Federation name set by the leader
+    pub federation_name: Option<String>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Encodable, Decodable)]
+/// Connection information sent between peers in order to start config gen
+pub struct PeerConnectionInfo {
+    /// TLS cert is necessary for P2P auth during DKG and consensus
+    pub cert: Vec<u8>,
+    /// P2P is the network for running DKG and consensus
+    pub p2p_url: SafeUrl,
+    /// API for secure websocket requests
+    pub api_url: SafeUrl,
+    /// Name of the peer, used in TLS auth
+    pub name: String,
+    /// Federation name set by the leader
+    pub federation_name: Option<String>,
+}
+
+impl PeerConnectionInfo {
+    pub fn encode_base58(&self) -> String {
+        format!(
+            "fedimint{}",
+            bs58::encode(&self.consensus_encode_to_vec()).into_string()
+        )
+    }
+
+    pub fn decode_base58(s: &str) -> anyhow::Result<Self> {
+        ensure!(s.starts_with("fedimint"), "Invalid Prefix");
```

Comment:
Consider triming (leading and trailing whitespace) and removing newlines. Ideally UI would sanitize it, but will it? People copy these over phone touch screens and emails where it might introduce line breaks etc.